File: 백준/Bronze/24356. ЧАСОВНИК/ЧАСОВНИК.py
t1, m1, t2, m2 = map(int, input().split())

start = t1 * 60 + m1
end = t2 * 60 + m2

if end >= start:
    m = end - start
else:
    m = (24 * 60 - start) + end

n = m // 30

print(m, n)

File: 백준/Bronze/24356. ЧАСОВНИК/ЧАСОВНИК.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int t1 = Integer.parseInt(st.nextToken());
        int m1 = Integer.parseInt(st.nextToken());
        int t2 = Integer.parseInt(st.nextToken());
        int m2 = Integer.parseInt(st.nextToken());

        int start = t1 * 60 + m1;
        int end = t2 * 60 + m2;

        int m = (end >= start)
                ? end - start
                : (24 * 60 - start) + end;

        int n = m / 30;

        System.out.println(m + " " + n);
    }
}